main/midi/midiUsb.c

```diff
@@ -0,0 +1,162 @@
+#include "midiUsb.h"
+
+#include <esp_log.h>
+
+#include "swadge2024.h"
+#include "tinyusb.h"
+
+//==============================================================================
+// Defines
+//==============================================================================
+
+// Define the total length of the MIDI USB Device Descriptorg
+#define MIDI_CONFIG_TOTAL_LEN (TUD_CONFIG_DESC_LEN + TUD_MIDI_DESC_LEN)
+
+//==============================================================================
+// Enums
+//==============================================================================
+
+// Interface counter
+enum interface_count
+{
+    ITF_NUM_MIDI = 0,
+    ITF_NUM_MIDI_STREAMING,
+
+    ITF_COUNT
+};
+
+// USB Endpoint numbers
+enum usb_endpoints
+{
+    // Available USB Endpoints: 5 IN/OUT EPs and 1 IN EP
+    EP_EMPTY = 0,
+    EPNUM_MIDI,
+};
+
+// Static function declarations
+static bool handlePacket(midiEvent_t* event, const uint8_t packet[4]);
+
+//==============================================================================
+// Variables
+//==============================================================================
+
+/**
+ * @brief MIDI Device String descriptor
+ */
+static const char* midiStringDescriptor[5] = {
+    // array of pointer to string descriptors
+    (char[]){0x09, 0x04},    // 0: is supported language is English (0x0409)
+    "MAGFest",               // 1: Manufacturer
+    "Swadge Synthesizer",    // 2: Product
+    "123456",                // 3: Serials
```

Comment:
Amazing, I have the same combination on my luggage!

Reply:
Genius!